WordPress/Classes/ViewRelated/Pages/PageListViewController.swift

```diff
@@ -755,13 +755,19 @@ class PageListViewController: AbstractPostListViewController, UIViewControllerRe
         let selectedPage = pageAtIndexPath(index)
         let newIndex = _tableViewHandler.index(for: selectedPage)
         let pages = _tableViewHandler.removePage(from: newIndex)
-        let parentPageNavigationController = ParentPageSettingsViewController.navigationController(with: pages, selectedPage: selectedPage) {
+        let parentPageNavigationController = ParentPageSettingsViewController.navigationController(with: pages, selectedPage: selectedPage, onClose: {
             self._tableViewHandler.isSearching = false
             self._tableViewHandler.refreshTableView(at: index)
-        }
+        }, onSuccess: { self.handleSetParentSuccess() } )
```

Comment:
I would suggest to use a [weak reference](https://www.avanderlee.com/swift/weak-self/) here and reformat the code like the following:
```
        let parentPageNavigationController = ParentPageSettingsViewController.navigationController(with: pages, selectedPage: selectedPage, onClose: { [weak self] in
            self?._tableViewHandler.isSearching = false
            self?._tableViewHandler.refreshTableView(at: index)
        }, onSuccess: { [weak self] in
            self?.handleSetParentSuccess()
        })
```

Reply:
Thanks for the info about weak reference. ❤️